pkg/k8s/watchers/pod.go

```diff
@@ -400,7 +405,9 @@ func (k *K8sWatcher) updateK8sPodV1(oldK8sPod, newK8sPod *slim_corev1.Pod) error
 	}
 
 	for _, podEP := range podEPs {
-		if labelsChanged {
+		if labelsChanged || uidChanged {
+			// Consider a UID change the same as a label change in case the pod's
+			// identity needs to be updated, see GH-30409.
```

Comment:
Please leave comments on why it's not necessary to do the same for when the annotations change.

Reply:
I'll update the comment. The short story is that annotations don't affect identities, so we don't need to trigger the logic for updating labels if the annotations changed.